### ISSUE TITLE
COPY fails on high number of shard count

### ISSUE THREAD

**Issue by metdos** [citusdata/citus#710]
Copy fails on a high number of shard count with a weird error message.

``` sql
CREATE TABLE lineitem
(
    l_orderkey    BIGINT not null,
    l_partkey     INTEGER,
    l_suppkey     INTEGER,
    l_linenumber  INTEGER,
    l_quantity    DOUBLE PRECISION,
    l_extendedprice  DOUBLE PRECISION,
    l_discount    DOUBLE PRECISION,
    l_tax         DOUBLE PRECISION,
    l_returnflag  CHAR(1),
    l_linestatus  CHAR(1),
    l_shipdate    DATE,
    l_commitdate  DATE,
    l_receiptdate DATE,
    l_shipinstruct CHAR(25),
    l_shipmode     CHAR(10),
    l_comment      VARCHAR(44)
);

SELECT master_create_distributed_table('lineitem', 'l_orderkey', 'hash');
SELECT master_create_worker_shards('lineitem', 1024, 1);
```

``` sql
COPY lineitem from '/home/ubuntu/tpch_2_13_0/lineitem.tbl' WITH DELIMITER '|';

WARNING:  connection failed to ip-10-218-175-197.ec2.internal:5432
DETAIL:  could not translate host name "ip-10-218-175-197.ec2.internal" to address: System error
CONTEXT:  COPY lineitem, line 9099: "9094|57706|5222|1|32|53238.40|0.04|0.03|N|O|1998-07-09|1998-08-19|1998-07-30|TAKE BACK RETURN|MAIL|y..."
ERROR:  could not find any active placements
CONTEXT:  COPY lineitem, line 9099: "9094|57706|5222|1|32|53238.40|0.04|0.03|N|O|1998-07-09|1998-08-19|1998-07-30|TAKE BACK RETURN|MAIL|y..."
```

**Comment by metdos** [comment]
Related to connection handling in #704.

**Comment by ozgune** [comment]
I'm also noting #484 and #617 as related issues.

**Comment by marcocitus** [comment]
Fixed by #3841
